File: IAIso-v5.0/core/iaiso-ruby/lib/iaiso/identity/scope_mapping.rb
# frozen_string_literal: true

module IAIso
  module Identity
    # Configures how OIDC claims become IAIso scopes.
    class ScopeMapping
      attr_reader :direct_claims, :group_to_scopes, :always_grant

      def initialize(direct_claims: [], group_to_scopes: {}, always_grant: [])
        @direct_claims = direct_claims.map(&:to_s).freeze
        @group_to_scopes = group_to_scopes.transform_values { |v| Array(v).map(&:to_s) }.freeze
        @always_grant = always_grant.map(&:to_s).freeze
        freeze
      end

      def self.defaults = new
    end
  end
end

File: IAIso-v5.0/core/iaiso-java/iaiso-identity/src/main/java/io/iaiso/identity/ScopeMapping.java
package io.iaiso.identity;

import java.util.Collections;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

/** Configures how OIDC claims become IAIso scopes. */
public final class ScopeMapping {
    public final List<String> directClaims;
    public final Map<String, List<String>> groupToScopes;
    public final List<String> alwaysGrant;

    public ScopeMapping(List<String> directClaims,
                        Map<String, List<String>> groupToScopes,
                        List<String> alwaysGrant) {
        this.directClaims = Collections.unmodifiableList(
            directClaims != null ? directClaims : Collections.emptyList());
        this.groupToScopes = Collections.unmodifiableMap(
            groupToScopes != null ? new HashMap<>(groupToScopes) : new HashMap<>());
        this.alwaysGrant = Collections.unmodifiableList(
            alwaysGrant != null ? alwaysGrant : Collections.emptyList());
    }

    public static ScopeMapping defaults() {
        return new ScopeMapping(null, null, null);
    }
}
